### PR TITLE
[core] Reload geojson source on url change

### LINKED ISSUE #6757: GeoJsonSource.setURL doesn't update source
Capturing from https://github.com/mapbox/mapbox-gl-native/pull/6751:

>    GeoJsonSource.setUrl(url)

Doesn't update the source, setting geojson itself works though.

cc @ivovandongen

### PR DESCRIPTION
**Pull request by ivovandongen**
Fixes #6757 

Review @tmpsantos  / @brunoabinader ?

### CODE REVIEW

File: src/mbgl/style/sources/geojson_source_impl.cpp [hunk 1]
```diff
@@ -132,6 +138,7 @@ void GeoJSONSource::Impl::loadDescription(FileSource& fileSource) {
 
             loaded = true;
             observer->onSourceLoaded(base);
+            req.reset();
```
Comment (jfirebaugh) [inline src/mbgl/style/sources/geojson_source_impl.cpp]: What's the rationale for this change?

File: src/mbgl/style/style.cpp [hunk 2]
```diff
@@ -491,6 +491,13 @@ void Style::onSourceError(Source& source, std::exception_ptr error) {
     observer->onResourceError(error);
 }
 
+void Style::onSourceDescriptionChanged(Source& source) {
+    observer->onSourceDescriptionChanged(source);
+    if (!source.baseImpl->loaded) {
+	   source.baseImpl->loadDescription(fileSource);
```
Comment (jfirebaugh) [inline src/mbgl/style/style.cpp]: Fix indentation here.

File: test/style/source.test.cpp [hunk 3]
```diff
@@ -381,3 +381,37 @@ TEST(Source, RasterTileAttribution) {
...
+
+    test.observer.tileError = [&] (Source&, const OverscaledTileID&, std::exception_ptr) {
+        FAIL() << "Should never be called";
+    };
+
+    GeoJSONSource source("source");
+    source.baseImpl->setObserver(&test.observer);
+    
+	//Load initial, so the source state will be loaded=true
+	source.baseImpl->loadDescription(test.fileSource);
```
Comment (jfirebaugh) [inline test/style/source.test.cpp]: Fix indentation here.

File: src/mbgl/style/sources/geojson_source_impl.cpp (hunk 1 above)
Comment (ivovandongen) [inline src/mbgl/style/sources/geojson_source_impl.cpp]: This enables us to re-request the description when the previous request completed (main use case was: https://github.com/mapbox/mapbox-gl-native/pull/6751). 

Alternatively we could clear the request on [setURL](https://github.com/mapbox/mapbox-gl-native/blob/35e6e41873f64c33880140d7396bf9fe366e64c0/src/mbgl/style/sources/geojson_source_impl.cpp#L39).

**Comment by ivovandongen** [comment]
@jfirebaugh Made the formatting changes in https://github.com/mapbox/mapbox-gl-native/pull/6783. Let me know if changes are needed for https://github.com/mapbox/mapbox-gl-native/pull/6770#discussion_r84344053 as well.

File: src/mbgl/style/sources/geojson_source_impl.cpp (hunk 1 above)
Comment (jfirebaugh) [inline src/mbgl/style/sources/geojson_source_impl.cpp]: Yeah, it should be done in `setURL`. Resetting here will disable automatic refreshes based on the caching headers, which is necessary to support things like traffic layers.
